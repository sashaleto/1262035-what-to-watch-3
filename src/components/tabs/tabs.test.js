import React from "react";
import renderer from "react-test-renderer";
import Tabs from "./tabs.jsx";

const movie = {
  title: `Gangs of new york`,
  posterImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
  previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
  backgroundImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
  description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
  rating: {
    score: 8.8,
    count: 370881,
  },
  director: `Martin Scorsese`,
  starring: [`Leonardo DiCaprio`, `Cameron Diaz`, `Daniel Day-Lewis`],
  genre: `Crime`,
  released: 2002,
  id: 1,
};

const comments = [{
  id: 1,
  user: {
    id: 18,
    name: `Sophie`
  },
  rating: 9,
  comment: `I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.`,
  date: `2020-03-05T17:15:47.441Z`
}, {
  id: 2,
  user: {
    id: 10,
    name: `Max`
  },
  rating: 4.4,
  comment: `A movie that will take you to another world full of emotions.`,
  date: `2020-02-28T17:15:47.441Z`
}];

it(`Render Tabs`, () => {
  const tree = renderer
        .create(<Tabs movie={movie} activeItem={null} onActivateItem={() => {}} comments={comments}/>)
        .toJSON();

  expect(tree).toMatchSnapshot();
});
