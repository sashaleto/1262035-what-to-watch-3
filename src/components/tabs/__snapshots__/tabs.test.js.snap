// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Tabs 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      8.8
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Very good
      </span>
      <span
        className="movie-rating__count"
      >
        370881
         ratings
      </span>
    </p>
  </div>
  <div
    className="movie-card__text"
  >
    <p>
      In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Martin Scorsese
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Leonardo DiCaprio, Cameron Diaz, Daniel Day-Lewis
         and other
      </strong>
    </p>
  </div>
</div>
`;
